src/commands/iam.ts

```diff
@@ -0,0 +1,79 @@
+import { ChatInputCommand, Command } from "@sapphire/framework";
+import { EmbedBuilder, GuildMemberRoleManager } from "discord.js";
+import { Logger } from "winston";
+import { Bot } from "..";
+import { EveLogger } from "../logging/logInitializer";
+import { EveLogUploader } from "../logging/logUploader";
+
+/**
+ * Command: /iam [course]
+ * Purpose: The bot tries to find the course referred and assigns the role to the user. Find course using: /[a-z]{4}\d{3}/gi (i.e., 4 letters + 3 numbers, case insensitive)
+ */
+export class IAMCommand extends Command {
+	public static readonly INVALID_ID = "_INVALID_";
+	public static readonly INVALID_COMMENT = '## Better improve the autocompletion results by typing more in the selection box.';
+	private logger?: Logger;
+    public constructor (context: Command.LoaderContext, options: Command.Options) {
+        super(context, {...options})
+    }
+
+    public override async registerApplicationCommands(registry: ChatInputCommand.Registry) {
+		// this.logger is a customised Winston Logger. It logs onto the console and onto Winston (file system) at the same time.
+		this.logger = (new EveLogger('iam')).cmd_logger;
```

Comment:
you are missing alot of code in this function that I put into the other commands in EVE to ensure each launch of the bot results in a fresh update of al the commands, specifically `ApplicationCommandRegistry`.

Reply:
Newest commit addresses this problem